### PR TITLE
MISRA Rules 10.4 and 10.7 on MQTT library files and iot_atomic_gcc.h

### PR DESCRIPTION
**Pull request by sarenameas**
*Issue #, if available:* https://issues.amazon.com/issues/TS-10708

*Description of changes:*
MISRA Rules 10.4 and 10.7 on MQTT library files and iot_atomic_gcc.h.

This rule updates the comparison of unsigned numbers.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/src/iot_mqtt_serialize.c [hunk 1]
```diff
@@ -722,7 +722,7 @@ static bool _subscriptionPacketSize( IotMqttOperationType_t type,
         /* Only SUBSCRIBE packets include the QoS. */
         if( type == IOT_MQTT_SUBSCRIBE )
         {
-            subscriptionPacketSize += 1;
+            subscriptionPacketSize += 1U );
```
Comment (eidelmanjonathan) [inline libraries/standard/mqtt/src/iot_mqtt_serialize.c]: You have an extra bracket here

Comment (sarenameas) [reply libraries/standard/mqtt/src/iot_mqtt_serialize.c]: Saw it.

File: libraries/standard/mqtt/src/iot_mqtt_operation.c [hunk 2]
```diff
@@ -760,7 +760,7 @@ void _IotMqtt_ProcessKeepAlive( IotTaskPool_t pTaskPool,
             swapStatus = Atomic_CompareAndSwap_u32( &( pPingreqOperation->u.operation.periodic.ping.failure ),
                                                     1,
                                                     0 );
-            IotMqtt_Assert( swapStatus == 1 );
+            IotMqtt_Assert( swapStatus == 1U );
```
Comment (muneebahmed10) [inline libraries/standard/mqtt/src/iot_mqtt_operation.c]: Curious why line 763 needs a `U` but lines 761 and 762 don't

Comment (sarenameas) [reply libraries/standard/mqtt/src/iot_mqtt_operation.c]: MISRA only called out the comparisons.

**Pull request merged**